File: basic/week(4)/fun_arr.js
// function functionName(parameter1, parameter2, ...) {
//     return value; 
// }

function add(a, b) {
    return a + b;
}
function createObject() {
    return { name: "John", age: 25 };
}
function getArray() {
    return [1, 2, 3];
}
function getFunction() {
    return function (x) {
        return x * x;
    };
}
function doNothing() {
    console.log("This does nothing!");
}

// Array 
// dynamic and not stored in contiguous memory locations. 
// Instead, they are objects with key-value pairs where keys are indices.
// Memory addresses are abstracted, so you cannot directly calculate memory locations.

let age = [10, 14, 16, 18, 19];

console.log("First element:", age[0]);
console.log("Second element:", age[1]);
console.log("Third element:", age[2]);

File: basic/week(4)/fun_arr.cpp
#include<iostream>
using namespace std;


// return_type function_name(parameter1, parameter2, ...) {
//     return value; 
// }
int add(int a, int b) {
    return a + b;
}
int* getPointer(int& a) {
    return &a;
}
class MyClass {
public:
    int value;
};

MyClass createObject() {
    MyClass obj;
    obj.value = 10;
    return obj;
}
void printMessage() {
    std::cout << "Hello, World!" << std::endl;
}
template <typename T>
T multiply(T a, T b) {
    return a * b;
}



void printAge(int age[], int n) {
    for (int i = 0; i < n; i++) {
        cout << age[i] << endl;
    }
}


int main() {
    int age[] = {10, 14, 16, 18, 19};
    cout << "Address of first element: " << &age[0] << endl;
    cout << "Address of second element: " << &age[1] << endl;
    cout << "Address of third element: " << &age[2] << endl;

    int myArray[] = {5, 4, 3, 2, 1};
    int size = sizeof(myArray); // Total size in bytes
    int numElements = sizeof(myArray) / sizeof(myArray[0]); // Number of elements
    cout << "Total size in bytes: " << size << endl;
    cout << "Number of elements: " << numElements << endl;
    
    int age[5] = {11, 14, 15, 18, 20};
    printAge(age, 5);
    return 0;

    return 0;
}